File: Data Structures & Algorithms/products-of-array-discluding-self/submission-9.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix = 1
        res = [1] * len(nums)
        for x in range(len(nums)):
            res[x] = prefix

            prefix *= nums[x]
        print(res)

        
        postfix = 1
        for x in range(len(nums)-1,-1,-1):
            res[x] *= postfix
            postfix *= nums[x]
        
        return res
    
        

File: Data Structures & Algorithms/two-integer-sum/submission-9.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();

        for(int i = 0; i < nums.length; i++)
        {
            int counterPart = target - nums[i];

            if(map.containsKey(counterPart))
            {

                int[] res = {map.get(counterPart), i};
                return res;
            }

            map.put(nums[i], i);
        }
        return new int[] {};
    }

    

}
